FIX: langfuse_v3 pipeline

rgaricano opened the pull request:
Revision of the pipeline, clean and fixed some issues.

rgaricano committed:
FIX: langfusev3 pipeline

Revision of the pipeline, clean and fixed some issues and v3 requirements.

rgaricano committed:
Modify  date in langfuse_v3_filter_pipeline.py

Updated author and date information in the Langfuse filter pipeline.

YetheSamartaka commented:
Hi @rgaricano . I tried applying your changes to my variant of the function filter (https://github.com/open-webui/functions/pull/11) but it performed worse as some traces were not traced at all or token counts were wrong. Where did you find that it has to be named "usage_details" instead of just "usage"?

rgaricano commented:
@YetheSamartaka,
https://github.com/open-webui/pipelines/pull/580
https://python.reference.langfuse.com/langfuse/api#Usage
https://python.reference.langfuse.com/langfuse/api#Observation.usage_details

Btw, In my PR I removed the `"total": input_tokens + output_tokens,` because it is done automaticatly by langfuse, (same for  `"unit": "TOKENS",`)

rgaricano commented:
@YetheSamartaka,
qq,
do you know if we can have the situation where only one input_tokens|output_tokens are None?
https://github.com/open-webui/pipelines/blob/14c2803a5d8e14a54abd6e01d43aaa4df0c6e006/examples/filters/langfuse_v3_filter_pipeline.py#L312

Maybe when streaming or when some error occurs during generation?
(In those cases the tokens update it's not going to be done)

I updated the PR for handle these cases, also I restored the total field (just in case).

rgaricano committed:
Enhance token usage extraction logic

Modified the usage_details logic to handle the cases where only one of input_tokens or output_tokens are None or missing in certain scenarios.

YetheSamartaka commented:
Thank you

rgaricano committed:
Update langfuse_v3_filter_pipeline.py

YetheSamartaka commented:
Hi. Perhaps you would want to take a look at this PR of mine as I have the Langfuse as a function. The pipeline itself is not always working correctly, mainly the issue is when user closes the chat window while the chat is being generated, the Langfuse will never get the data properly.
https://github.com/open-webui/functions/pull/11/files#diff-81f33f1799933618031da975bc0b32306c7736ec23b78dc8ca3ca29a6698a1b3

rgaricano committed:
Update langfuse requirement version to 3.2

As @Chrisadas noted in https://github.com/open-webui/pipelines/issues/556#issuecomment-4697474697
this requirement with lower langfuse versions costs isn't catched correctly.
